librad/src/meta/entity.rs

```diff
@@ -198,16 +198,12 @@ impl EntityTimestamp {
     }
 }
 
-impl Into<SystemTime> for EntityTimestamp {
-    fn into(self) -> SystemTime {
-        if self.0 >= 0 {
-            UNIX_EPOCH
-                .checked_add(Duration::from_millis(self.0 as u64))
-                .unwrap()
+impl From<EntityTimestamp> for SystemTime {
+    fn from(timestamp: EntityTimestamp) -> Self {
+        if timestamp.0 >= 0 {
+            UNIX_EPOCH + Duration::from_millis(timestamp.0 as u64)
```

Comment:
So the issue here is that we may overflow/underflow now, no?

Reply:
The issue is that there's no concept of negative `Duration`.
To be fair, this is not apparent in `EntityTimestamp::current_time` because of the assumption that `SystemTime::now()` is _always_ after the Unix epoch... right now it would silently produce a wrong timestamp (set to the epoch, exactly).

Should we handle system clocks set before the epoch?
Or panic explicitly?

Reply:
Time showing it's hairy head already :grin: While not being sure about the merits of time overall; if you were to handle it I think something similar to @kim's suggestion for when I was working on issues could work better https://github.com/radicle-dev/radicle-link/blob/5b524cb4849c5f57dc72766222d06bc3046811f4/radicle-tracker/src/metadata/clock.rs.

Actually, why are we converting in the first place? :thinking:

Reply:
I fixed it to handle system clocks set before the epoch, after all it's a one liner.

About "why converting", it is to give the user some means to view a "timestamp" as a moment in time using the standard Rust facilities to handle moments in time (`SystemTime`).

The `EntityTimestamp` type exists only as a convenience to have a compact, well understood and language-independent serialization format (a simple integer, millis from epoch), and for convenience with direct comparisons (to be able to do signed comparisons if needed, which `SystemTime` does not support).